pkg/identityserver/end_device_registry.go

```diff
@@ -228,6 +243,26 @@ func (is *IdentityServer) updateEndDevice(ctx context.Context, req *ttnpb.Update
 	return dev, nil
 }
 
+var errMultipleFieldBatchUpdate = errors.DefineInvalidArgument(
+	"multiple_update_fields_specified",
+	"multiple fieldmask paths specified while batch update allows only one field update",
+)
+
+func (is *IdentityServer) batchUpdateEndDeviceLastSeen(ctx context.Context, req *ttnpb.BatchUpdateEndDeviceLastSeenRequest) (*ttnpb.EndDevices, error) {
+	if err := clusterauth.Authorized(ctx); err != nil {
+		return nil, err
+	}
+	endDevices := make([]*ttnpb.EndDevice, len(req.EndDevices.EndDevices))
+	err := is.store.Transact(ctx, func(ctx context.Context, st store.Store) (err error) {
+		endDevices, err = st.BatchUpdateEndDeviceLastSeen(ctx, req.GetEndDevices().EndDevices)
```

Comment:
Why use a getter if you will access the field directly afterwards ? It would panic for `nil`.